packages/sanity/src/structure/comments/src/hooks/useCommentsUpsell.ts

```diff
@@ -0,0 +1,30 @@
+import {useContext} from 'react'
+import {CommentsUpsellContext} from '../context'
+import {CommentsUpsellContextValue} from '../context/upsell/types'
+
+/**
+ * @beta
+ * @hidden
+ */
+export function useCommentsUpsell(): CommentsUpsellContextValue {
+  const value = useContext(CommentsUpsellContext)
+
+  if (!value) {
+    // Instead of throwing, we return a dummy value to avoid breaking the CommentsField implementation, given the context is optional.
```

Comment:
Might be nice to add a `console.error` here to notify future devs in case the provider moves or some other implementation causes this to break

Reply:
We can't add a `console.error` here because the context will be undefined when you are not in upsell mode. So all the users with the feature properly enabled will see the `console.error` added here